mavlink/test_mission_ended.py

```diff
@@ -0,0 +1,137 @@
+"""
+Test for drone's destination at final waypoint by uploading mission and monitoring it.
+"""
+import sys
+import time
+
+import dronekit
+
+from mavlink.modules import flight_controller
+
+
+DELAY_TIME = 1.0  # seconds
+MISSION_PLANNER_ADDRESS = "tcp:127.0.0.1:14550"
+TIMEOUT = 1.0  # seconds
+
+MAVLINK_TAKEOFF_FRAME = dronekit.mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT
+MAVLINK_TAKEOFF_COMMAND = dronekit.mavutil.mavlink.MAV_CMD_NAV_TAKEOFF
+MAVLINK_FRAME = dronekit.mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT
+MAVLINK_COMMAND = dronekit.mavutil.mavlink.MAV_CMD_NAV_WAYPOINT
+
+ALTITUDE = 10  # metres
+ACCEPT_RADIUS = 10  # metres
+
+
+# TODO: This function is to be removed when Dronekit-Python interfaces are
+# moved from pathing repository.
+def upload_mission(controller: flight_controller.FlightController,
+                   waypoints: "list[tuple[float, float, float]]") -> bool:
+    """
+    Add a takeoff command and waypoint following commands to the drone's
+    command sequence, and upload them.
+
+    Parameters
+    ----------
+    controller: "flight_controller.FlightController"
+    waypoints: "list[tuple[float, float, float]]"
```

Comment:
Explain what the tuple actually means beyond being an ordered list of numbers (e.g. latitude, longitude, altitude). Include units.